causalml/feature_selection/filters.py

```diff
@@ -0,0 +1,370 @@
+
+        Parameters
+        ----------
+        data : DataFrame
+            The DataFrame that contains all the data (in the current "node").  
+            - [TODO] We may need a index vector to restrict the rows of data 
+              outside the function.
+            - [TODO] Currently the positions of treatment_group_key and 
+              conversion in data are hard coded.  Modify the code for more
+              flexibility.
+
+        Returns
+        -------
+        results : dict
+            Counts of conversions by treatment groups, of the form: 
+            {'control': {0: 10, 1: 8}, 'treatment1': {0: 5, 1: 15}}
+        nodeSummary: dict
+            Probability of conversion and group size by treatment groups, of 
+            the form:
+            {'control': [0.490, 500], 'treatment1': [0.584, 500]}
+
+        References
+        ----------
+        - results: upliftpy.UpliftTreeClassifier.group_uniqueCounts() 
+        - nodeSummary: upliftpy.UpliftTreeClassifier.tree_node_summary()
+        """
+
+        # Note: results and nodeSummary are both dict with treatment_group_key
+        # as the key.  So we can compute the treatment effect and/or 
+        # divergence easily.
+
+        # Counts of conversions by treatment group
+        results_series = data.groupby([experiment_group_column, y_name]).size()
+        
+        treatment_group_keys = results_series.index.levels[0].tolist()
+        y_name_keys = results_series.index.levels[1].tolist()
+
+        results = {}
+        for ti in treatment_group_keys: 
+            results.update({ti: {}}) 
+            # results[ti] = {} # alternatively
+            for ci in y_name_keys:
+                results[ti].update({ci: results_series[ti, ci]}) 
+                # results[ti][ci] = results_series[ti, ci] # alternatively
+
+        # Probability of conversion and group size by treatment group
+        nodeSummary = {}
+        for treatment_group_key in results: 
+            n_1 = results[treatment_group_key][1]
+            n_total = (results[treatment_group_key][1] 
+                       + results[treatment_group_key][0])
+            y_mean = 1.0 * n_1 / n_total
+            nodeSummary[treatment_group_key] = [y_mean, n_total]
+        # [TODO] More complicated situations in tree_node_summary()
+        
+        return results, nodeSummary 
+
+    # Divergence-related functions, from upliftpy
+    def _kl_divergence(self, pk, qk):
+        """
+        Calculate KL Divergence for binary classification.
+
+        Parameters
+        ----------
+        pk, qk : float
+        """
+        # Formula
+        # S = sum(np.array(pk) * np.log(np.array(pk) / np.array(qk)))
+        if qk < 0.1**6:
+            qk = 0.1**6
+        elif qk > 1 - 0.1**6:
+            qk = 1 - 0.1**6
+        S = pk * np.log(pk / qk) + (1-pk) * np.log((1-pk) / (1-qk))
+        return S
+
+    def _evaluate_KL(self, nodeSummary, control_group='control'):
```

Comment:
Same question as above regarding the default argument. I think we use 0 as the default control label elsewhere.

Reply:
The current data generation function generates 'control' and 'treatment', and this setting would compile with the DGP well. However, this is an intermediate function, that is not user-facing.